File: projinfo.py
#!/usr/bin/env python3
import sys
from argparse import ArgumentParser #take in sys arg
from helpers import *
    #proj_users, pi_projs, semester, proj_pi, user_pis

def main(): #parse arguments, returns input in a dictionary
    '''
    once i add more queries i think i'll create a dictionary of inputs and corresponding outputs
    currently 
    '''
    #create parser + define arguments
    parser = ArgumentParser(description='tool to return information about current projects and PIs') #not sure where this description will show up so idk what to put here lol
    parser.add_argument('-a', '--academic', action='store_true', help='No input necessary')
    parser.add_argument('-u', '--user', help='User login name')
    parser.add_argument('-p', '--project', help='Projects info')
    parser.add_argument('-pi', '--lpi', help='PI login name')
    parser.add_argument('-v', '--v', action='store_true', help="add to -u or -p for additional info")


    #create a dictionary of all the flags, not used 
    #inputs = vars(args)  #this should automatically set the flags with no input to None

    #process the flags and print their outputs?
    #multiple inputs will only process the first one. 
    input1 = True

    #if len(sys.argv) == 1:

    if len(sys.argv) == 1:
        parser.print_help()
        sys.exit()

    args = parser.parse_args()
    if args.academic:
        print(academic())

    for i in range(1, len(sys.argv)-1):  # Start from index 1 to skip script name
        arg = sys.argv[i]

        if arg.startswith('-'):
            if input1 == False:
                print("All secondary inputs will be ignored")

            flag = arg.lstrip('-')
            
            value = sys.argv[i+1]

            if value is not None and input1 == True:
                if flag == ('p' or 'project'):
                    print(proj(value))
                if flag == ('u' or 'user'):
                    if len(sys.argv) > (i+2):
                        if sys.argv[i+2] == '-v':
                            print(vuser_pis(value))
                    else:    
                        print(user_pis(value))
                if flag == ('pi' or 'lpi'):
                    print(pi_projs(value))
                input1 = False


if __name__ == "__main__":
    main()










''' 
def main():
    '''
'''
    adding the arg parse code here!
    - use dictionary to map inputs to functions
    '''
'''
    #create parser + define arguments
    parser = ArgumentParser(description='tool to return information about current projects and PIs') #not sure where this description will show up so idk what to put here lol
    parser.add_argument('function', help='function name: proj_users, pi_projs, semester')
    parser.add_argument('-i', '--input', help='input for functions that require it: project name for proj_users, PI name for pi_projs')

    #take in command line arguments
    args = parser.parse_args()

    #save the arguments
    function = args.function
    #print('input function:', function)

    if args.input:
        func_input = args.input
        #print('input func input:', func_input)

    #dictionary to map different functions
    functions = {
        'proj_users': proj_users,
        'pi_projs': pi_projs,
        'semester': semester
    }

    call = functions.get(function) #use .get() to avoid KeyError

    if call:
        if 'func_input' in locals(): #check for secondary input
            call(func_input)
        else:
            call()
    else:
        print("invalid input")

if __name__ == '__main__':
    main()
'''

File: projinfo.pl
#!/usr/bin/env perl
use strict;
use warnings;
use FindBin;
use lib $FindBin::Bin;
use Getopt::Long qw(GetOptions :config auto_help);
use Pod::Usage;
use Data::Dumper;

# Import helper functions from helpers.pm (make sure this file exists)
require "helpers.pm";

# Initialize command line arguments
my $academic = 0;
my $user;
my $project;
my $lpi;
my $verbose = 0;

# Store the original number of command-line arguments
my $num_args = @ARGV;

# Parsing command line arguments
GetOptions(
    'a|academic' => \$academic, # '-a' or '-academic'
    'u|user=s' => \$user,
    'p|project=s' => \$project,
    'pi|lpi=s' => \$lpi,
    'v|verbose' => \$verbose,
) or die "Error in command line arguments\n";

# Check for file issues, assuming file_issues() is a function in Helpers.pm
if (Helpers::file_issues()) {
    print "The data file is empty\n";
    exit;
}

# No arguments provided, show help
if ($num_args == 0) {
    print "no arguments provided";
    pod2usage(-verbose => 1);
    exit;
}

# Process the academic flag
if ($academic) {
    print Helpers::academic();
    exit;
}

# Initialize a flag to check if the first argument has been processed
my $input1 = 1;

# Process other arguments
if ($project and $input1) {
    # print Helpers::proj($project);
    # $input1 = 0;
    
    my $project_info = Helpers::proj($project);  # Use the correct package name
    if (defined $project_info) {
        print "Project Information for '$project':\n$project_info\n";
    } else {
        print "No information found for project '$project'.\n";
    }
}
if ($user and $input1) {
    if ($verbose) {
        print Helpers::vuser_pis($user);
    } else {
        print Helpers::user_pis($user);
    }
}
if ($lpi and $input1) {
    if ($verbose) {
        print Helpers::vpi_projs($lpi);
    } else {
        print Helpers::pi_projs($lpi);
    }
}


__END__

=head1 NAME

projinfo.pl - Tool to return information about current projects and PIs.

=head1 SYNOPSIS

Usage: tool to return information about current projects and PIs

Options:

    -a, --academic        --academic    No input necessary, shows academic information.

    -u, --user=<name>     --user        User login name to get user-specific information.

    -p, --project=<name>  --project     Project name to get project-specific information.

    -pi, --lpi=<name>     --lpi         PI login name to get PI-specific information.

    -v, --verbose         --v           Add to -u or -p for additional information.

=head1 DESCRIPTION

This script provides information about projects, users, and PIs based on
the provided command-line options.

=cut